addon/templates/components/polaris-button.hbs

```diff
@@ -1,63 +1,80 @@
-{{#component
-  buttonComponentName
-  class=class
-  url=url
-  external=external
-  primary=primary
-  destructive=destructive
-  disabled=isDisabled
-  loading=loading
-  size=size
-  outline=outline
-  fullWidth=fullWidth
-  submit=submit
-  plain=plain
-  accessibilityLabel=accessibilityLabel
-  ariaControls=ariaControls
-  ariaExpanded=ariaExpandedValue
-  iconOnly=iconOnly
-  onClick=onClick
-  onFocus=onFocus
-  onBlur=onBlur
-  mouseUp=handleMouseUpByBlurring
-  data-test-id=dataTestId
+{{#with
+  (component "polaris-button/content"
+    text=text
+    primary=primary
+    destructive=destructive
+    loading=loading
+    disclosure=disclosure
+    icon=icon
+  )
+  as |contentComponent|
 }}
-  <span class="Polaris-Button__Content">
-    {{#if loading}}
-      <span class="Polaris-Button__Spinner">
-        {{polaris-spinner
-          size="small"
-          accessibilityLabel="Loading"
-          color=spinnerColor
-        }}
-      </span>
-    {{/if}}
-
-    {{#if icon}}
-      <span class="Polaris-Button__Icon">
-        {{#if (is-component-definition icon)}}
-          {{render-content icon}}
+  {{#if url}}
+    {{#if isDisabled}}
+      {{!--
+        Render an `<a>` so toggling disabled/enabled state changes only the
```

Comment:
is this really a problem in Ember...or is it more to keep implementation as close as possible?

Reply:
The latter - I don't see any real disadvantage to having this implemented and it makes sure we're matching the React implementation closely, which can't hurt 🤷‍♂️